src/apps/search-header/search-header.tsx

```diff
@@ -183,7 +203,10 @@ const SearchHeader: React.FC = () => {
   });
 
   return (
-    <form className="header__menu-second" action={t("searchHeaderUrlText")}>
+    <form
+      className="header__menu-second"
+      action={String(constructSearchUrl(searchUrl, qWithoutQuery))}
```

Comment:
that sets the query string (from a suggestion) into the search field without calling a new suggestion query. We use it here because it is more accurate than q.